metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1369,7 +1384,23 @@ void handleBrokerInControlledShutdown(int brokerId, long brokerEpoch, List<ApiMe
                 (short) 1));
         }
         generateLeaderAndIsrUpdates("enterControlledShutdown[" + brokerId + "]",
-            brokerId, NO_LEADER, records, brokersToIsrs.partitionsWithBrokerInIsr(brokerId));
+            brokerId, NO_LEADER, NO_LEADER, records, brokersToIsrs.partitionsWithBrokerInIsr(brokerId));
+    }
+
+    /**
+     * Generate the appropriate records to handle a broker registering after unclean shutdown.
```

Comment:
The first sentence here doesn't really make sense to me. The second sentence is more descriptive of what this method does, maybe we can just keep that one? Or something similar

> Create partition change records to remove replicas from any ISR or ELR for brokers doing unclean shutdown.